app/services/customers/create_service.rb

```diff
@@ -176,6 +176,12 @@ def handle_api_billing_configuration(customer, params, new_customer)
 
       return if customer.payment_provider.nil?
 
+      update_provider_customer = (billing || {})[:provider_customer_id].present?
+      update_provider_customer ||=
+        customer.__send__("#{customer.payment_provider}_customer")&.provider_customer_id.present?
```

Comment:
What about extracting this logic and avoiding the meta-programation here?

Something like:
```ruby
customer.provider_customer&.provider_customer_id.present?

def provider_customer # In Customer model?
  if payment_provider == 'stripe'
    stripe_customer
  elsif payment_provider == 'gocardless'
    gocardless_customer
  end
end
```

Reply:
updated

Reply:
Cool - we can add a unit test for the model as well no? 😅

Reply:
Tests, what? 😄 Added test cases!